xbmc/utils/StringUtils.cpp

```diff
@@ -16,7 +16,11 @@
 //
 //------------------------------------------------------------------------
 
+#ifdef TARGET_WINDOWS
+#include <guid.hpp>
+#else
 #include <guid.h>
+#endif
```

Comment:
no platform hacks, if newer versions ship `guid.hpp`

Reply:
feel free to fix the other platforms, that's not something I intend to do. There's enough work done in this PR

Reply:
Either use a version of crossguid which is compatible with the current code or adjust the code the right way, without hacks.